command/sub8_missions/missions/start_gate.py

```diff
@@ -21,96 +21,98 @@
-    fprint('Turning on vision service')
-    yield start_gate_enable(SetBoolRequest(data=True))
-
-    start_gate_search = yield sub_singleton.nh.get_service_client('/vision/start_gate/pose', VisionRequest2D)
-    fprint("Searching for start gate pose")
-    start_gate_search_res = yield start_gate_search(VisionRequest2DRequest(target_name=''))
-
-    if not start_gate_search_res.found:
-        fprint("Waiting a few seconds and trying again:")
-        time.sleep(3)
-        start_gate_search_res = yield start_gate_search(VisionRequest2DRequest(target_name=''))
-    # yield start_gate_enable(SetBoolRequest(data=False))
-    # yield start_gate_enable(SetBoolRequest(data=True))
-    if not start_gate_search_res.found:
-        fprint("Running search pattern")
-        global FOUND_START_GATE
-        while not FOUND_START_GATE and start_gate_search_res.pose.x == 0:
-            fprint(FOUND_START_GATE)
-            time.sleep(1)
-            fprint("AGAIN - Searching for start gate pose")
-            start_gate_find(sub_singleton)
-            yield search_pattern(sub_singleton)
-    
-    fprint("Found start gate: " + str(start_gate_search_res.pose))
-    yield align_for_dummies(sub_singleton, start_gate_search_res)
-    fprint("YOLO")
-
-    while(FOUND_START_GATE):
-        start_gate_find(sub_singleton)
-        yield sub_singleton.move.forward(.3).zero_roll_and_pitch().go(speed=SPEED)
-
-    fprint("Finished")
+class StartGateMission:
 
+    FOUND_START_GATE = False
+    SPEED = 0.3
 
-@txros.util.cancellableInlineCallbacks
-def start_gate_find(sub_singleton):
-    global FOUND_START_GATE
-    start_gate_search = yield sub_singleton.nh.get_service_client('/vision/start_gate/pose', VisionRequest2D)
-    start_gate_search_res = yield start_gate_search(VisionRequest2DRequest(target_name=''))
-    FOUND_START_GATE = start_gate_search_res.found
+    def __init__(self, sub):
+        self.sub_singleton = sub
 
-@txros.util.cancellableInlineCallbacks
-def search_pattern(sub_singleton):
-    if not FOUND_START_GATE:
-        fprint("Moving up .3", msg_color="yellow")
-        yield sub_singleton.move.up(.3).zero_roll_and_pitch().go(speed=SPEED)
-        fprint("Moving down .3", msg_color="yellow")
-        yield sub_singleton.move.down(.3).zero_roll_and_pitch().go(speed=SPEED)
-    if not FOUND_START_GATE:
-        fprint("Moving left .3", msg_color="yellow")
-        yield sub_singleton.move.left(.3).zero_roll_and_pitch().go(speed=SPEED)
-        fprint("Moving right .3", msg_color="yellow")
-        yield sub_singleton.move.right(.3).zero_roll_and_pitch().go(speed=SPEED)
+    @txros.util.cancellableInlineCallbacks
+    def run_mission(self):
+        fprint('Starting mission')
+        fprint('Assuming camera is facing gate')
+        fprint('Leveling off', msg_color="yellow")
+        yield self.sub_singleton.move.zero_roll_and_pitch().go()
+        fprint('Going down', msg_color="yellow")
+        yield self.sub_singleton.move.down(1).go()
```

Comment:
This seems arbitrary since you don't know at what depth you're starting, what's the reason?